postprocessing/pp-webm.c

```diff
@@ -75,92 +47,30 @@ int janus_pp_webm_create(char *destination, char *metadata, gboolean vp8) {
 		return -1;
 	}
 #endif
-	/* Setup FFmpeg */
-#if ( LIBAVFORMAT_VERSION_INT < AV_VERSION_INT(58,9,100) )
-	av_register_all();
-#endif
-	/* Adjust logging to match the postprocessor's */
-	av_log_set_level(janus_log_level <= LOG_NONE ? AV_LOG_QUIET :
-		(janus_log_level == LOG_FATAL ? AV_LOG_FATAL :
-			(janus_log_level == LOG_ERR ? AV_LOG_ERROR :
-				(janus_log_level == LOG_WARN ? AV_LOG_WARNING :
-					(janus_log_level == LOG_INFO ? AV_LOG_INFO :
-						(janus_log_level == LOG_VERB ? AV_LOG_VERBOSE : AV_LOG_DEBUG))))));
 	/* WebM output */
-	fctx = avformat_alloc_context();
+	fctx = janus_pp_create_avformatcontext("webm", metadata, destination);
 	if(fctx == NULL) {
 		JANUS_LOG(LOG_ERR, "Error allocating context\n");
 		return -1;
 	}
-	/* We save the metadata part as a comment (see #1189) */
-	if(metadata)
-		av_dict_set(&fctx->metadata, "comment", metadata, 0);
-	fctx->oformat = av_guess_format("webm", NULL, NULL);
-	if(fctx->oformat == NULL) {
-		JANUS_LOG(LOG_ERR, "Error guessing format\n");
-		return -1;
-	}
-    char filename[1024];
-	snprintf(filename, sizeof(filename), "%s", destination);
-#ifdef USE_CODECPAR
-	AVCodec *codec = avcodec_find_encoder(vp8 ? AV_CODEC_ID_VP8 : AV_CODEC_ID_VP9);
-	if(!codec) {
-		/* Error opening video codec */
-		JANUS_LOG(LOG_ERR, "Encoder not available\n");
-		return -1;
-	}
-	fctx->video_codec = codec;
-	fctx->oformat->video_codec = codec->id;
-	vStream = avformat_new_stream(fctx, codec);
-	vStream->id = fctx->nb_streams-1;
-	vEncoder = avcodec_alloc_context3(codec);
-	vEncoder->width = max_width;
-	vEncoder->height = max_height;
-	vEncoder->time_base = (AVRational){ 1, fps };
-	vEncoder->pix_fmt = AV_PIX_FMT_YUV420P;
-	vEncoder->flags |= CODEC_FLAG_GLOBAL_HEADER;
-	if(avcodec_open2(vEncoder, codec, NULL) < 0) {
-		/* Error opening video codec */
-		JANUS_LOG(LOG_ERR, "Encoder error\n");
-		return -1;
-	}
-	avcodec_parameters_from_context(vStream->codecpar, vEncoder);
-#else
-	//~ vStream = av_new_stream(fctx, 0);
-	vStream = avformat_new_stream(fctx, 0);
-	if(vStream == NULL) {
-		JANUS_LOG(LOG_ERR, "Error adding stream\n");
-		return -1;
-	}
-	//~ avcodec_get_context_defaults2(vStream->codec, CODEC_TYPE_VIDEO);
-#if LIBAVCODEC_VER_AT_LEAST(53, 21)
-	avcodec_get_context_defaults3(vStream->codec, AVMEDIA_TYPE_VIDEO);
-#else
-	avcodec_get_context_defaults2(vStream->codec, AVMEDIA_TYPE_VIDEO);
-#endif
+
+	int codec_id;
 #if LIBAVCODEC_VER_AT_LEAST(54, 25)
 	#if LIBAVCODEC_VERSION_MAJOR >= 55
-	vStream->codec->codec_id = vp8 ? AV_CODEC_ID_VP8 : AV_CODEC_ID_VP9;
+		codec_id = vp8 ? AV_CODEC_ID_VP8 : AV_CODEC_ID_VP9;
 	#else
-	vStream->codec->codec_id = AV_CODEC_ID_VP8;
+		codec_id = AV_CODEC_ID_VP8;
```

Comment:
The two `codec_id` lines don't need to be indented (the one below isn't, for instance), as macro `ifs` for us are not like "regular" ifs.